code/game/dna/dna_modifier.dm

```diff
@@ -533,7 +533,7 @@
 		occupantData["name"] = connected.occupant.dna.real_name
 		occupantData["stat"] = connected.occupant.stat
 		occupantData["isViableSubject"] = 1
-		if((NOCLONE in connected.occupant.mutations && connected.scan_level < 3) || !src.connected.occupant.dna)
+		if((NOCLONE in connected.occupant.mutations && connected.scan_level < 3) || !src.connected.occupant.dna || ismachine(connected.occupant))
```

Comment:
This check should not be `ismachine`, but rather `connected.occupant.dna.species.bodyflags & NO_DNA`)